pallets/baby-liminal/src/lib.rs

```diff
@@ -334,71 +342,60 @@ pub mod pallet {
             verification_key_identifier: VerificationKeyIdentifier,
             proof: Vec<u8>,
             public_input: Vec<u8>,
-        ) -> Result<(), (Error<T>, Option<Weight>)> {
-            Self::_bare_verify::<Groth16>(verification_key_identifier, proof, public_input)
-        }
-
-        fn _bare_verify<S: VerifyingSystem>(
-            verification_key_identifier: VerificationKeyIdentifier,
-            proof: Vec<u8>,
-            public_input: Vec<u8>,
         ) -> Result<(), (Error<T>, Option<Weight>)> {
             let data_length_limit = T::MaximumDataLength::get() as usize;
             let data_length_excess = proof.len().saturating_sub(data_length_limit)
```

Comment:
I don't understand this computation: specifically why subtract with saturation and but add both values with a "regular" arithmetic op?

Maybe use checked operations anyway, even if you need to introduce extra error variant